include/common/reduce_pattern.h

```diff
@@ -41,7 +41,7 @@ class reduce_t {
   \param Id Identity value.
   \param combine_op Combiner used for the reduction.
   */
-  reduce_t(int wsize, int offset, Identity id, Combiner && combine_op) :
+  reduce_t(size_t wsize, size_t offset, Identity id, Combiner && combine_op) :
```

Comment:
In any place where `size_t` is used. It should be `std::size_t`.

However, I disagree with using unsigned for sizes and offsets. Please see CppCoreGuidelines ES.100 to ES.107

Reply:
The issue relays on Line 68 and 78 of the same file. We are comparing an unsigned number (`.size()` aka `unsigned long`) with a signed number (aka `int`), clangs flags this as a warning. 
ES.100 recommends to avoid mixing them. It is true the second note kind of exclude this case, as the offset is a sort of difference of pointers but unless we plan on having negative offsets, a window size should always be positive.

Reply:
The current implementation of reduce_t (holding items in a vector) should not impact its interface.
However, I do agree that right now there is an issue in the current implementation with the comparison:

`items.size() >= window_size_`

The problem would arise when the number of elements in the reduction is greater than `std::numeric_limits<decltype(window_size_)>::max()`. This is only a theoretical issue because in that situation the reduction is useless and non-sense. 

Proposed resolution:
1.- Change the comparison to:
`static_cast<int>(items.size()) >= window_size_`

2.- Document a class invariant (in doxygen) stating that items.size() should always be below numeric_limits<int>::max(). Note that this invariant is true by design as `remaining_` max value is `offset_ - window_size_` and the difference is always to fit into an `int`

Reply:
I think the best option is to use the `static_cast` solution